src/coreclr/jit/lclvars.cpp

```diff
@@ -4590,6 +4592,8 @@ void Compiler::lvaMarkLocalVars()
         }
     }
 
+    bool addedLocal = false;
```

Comment:
It might be simpler to cache `lvaCount` here and then check `lvaCount` against the cached value at the end, instead of interspersing `addedLocal = true` calls and hoping you didn't miss any place.

Reply:
Yes, that would be more robust. Will do.